src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -155,5 +161,124 @@ private bool CheckReturnType(IdentToken returnType, List<TexlError> errors)
 
             return isReturnTypeCheckSuccessful;
         }
+
+// This code is intended as a prototype of the Partial attribute system, for use in solution layering cases
+// Provides order-independent ways of merging named formulas
+        #region Partial Attributes
+
+        private static readonly string _renamedFormulaGuid = Guid.NewGuid().ToString("N");
+
+        /// <summary>
+        /// For NamedFormulas with partial attributes,
+        /// validates that the same attribute is applied to all matching names,
+        /// then applies name mangling to all, and constructs a separate
+        /// formula with the operation applied and the original name.
+        /// </summary>
+        private ParseUserDefinitionResult ProcessPartialAttributes(ParseUserDefinitionResult parsed)
+        {
+            var groupedFormulas = parsed.NamedFormulas.GroupBy(nf => nf.Ident.Name.Value);
+            var errors = parsed.Errors?.ToList() ?? new List<TexlError>();
+            var newFormulas = new List<NamedFormula>();
+
+            foreach (var nameGroup in groupedFormulas)
+            {
+                var name = nameGroup.Key;
+                var firstAttribute = nameGroup.Select(nf => nf.Attribute).FirstOrDefault(att => att != null);
+
+                if (firstAttribute == null || nameGroup.Count() == 1)
+                {
+                    newFormulas.AddRange(nameGroup);
+                    continue;
+                }
+
+                var updatedGroupFormulas = new List<NamedFormula>();
```

Comment:
>var updatedGroupFormulas = new List<NamedFormula>(); [](http://example.com/codeflow?start=16&length=52)

You can initialize this outside foreach and just clear it here.

Reply:
Yes, but IMO this is clearer. That seems like an optimization that offers very little benefit, as this isn't a hot path.

Reply:
I don't see how it is clearer. It's extra allocation call per iteration which can be easily be avoided. But if you want to keep it this way then it's fine.